File: LeetCode Solutions/LeetCoding Challenge/May 2020 LeetCoding Challenge/Valid Number.py
class Solution:
    def isNumber(self, s: str) -> bool:
        num, exponent, sign, decimal = False, False, False, False
        for i in s:
            if(i >= '0' and i <= '9'):
                num = True
            elif(i == 'e' or i == 'E'):
                if(exponent or not num):
                    return False
                else:
                    exponent, num, sign, decimal = True, False, False, False
            elif(i == '+' or i == '-'):
                if(sign or num or decimal):
                    return False
                else:
                    sign = True
            elif(i == '.'):
                if(decimal or exponent):
                    return False
                else:
                    decimal = True
            else:
                return False
        return num

File: LeetCode Solutions/LeetCoding Challenge/May 2020 LeetCoding Challenge/Valid Number.java
class Solution {
    public boolean isNumber(String s) {
        boolean num = false, exponent = false, sign = false, decimal = false;
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            if(c >= '0' && c <= '9'){
                num = true;
            }
            else if(c == 'e' || c == 'E'){
                if(exponent || !num){
                    return false;
                } else{
                    exponent = true;
                    sign = false;
                    num = false;
                    decimal = false;
                }
            }
            else if(c == '+' || c == '-'){
                if(sign || num || decimal){
                    return false;
                }
                else{
                    sign = true;
                }
            }
            else if(c == '.'){
                if(decimal || exponent){
                    return false;
                }
                else{
                    decimal = true;
                }
            }
            else{
                return false;
            }
        }
        return num;
    }
}